src/main/java/dev/slimevr/vr/processor/skeleton/LegTweaks.java

```diff
@@ -0,0 +1,822 @@
+		}
+		if (bufferHead.getRightLegState() == LegTweakBuffer.LOCKED) {
+			rightFootPosition.x = bufferHead.getParent().getRightFootPositionCorrected().x;
+			rightFootPosition.z = bufferHead.getParent().getRightFootPositionCorrected().z;
+		}
+
+		// for either foot that is unlocked get its last position and calculate
+		// its position for this frame. the amount of displacement is based on
+		// the distance between the last position, the current position, and
+		// the hyperparameters
+		correctUnlockedLeftFootTracker();
+		correctUnlockedRightFootTracker();
+	}
+
+	private void correctUnlockedLeftFootTracker() {
+		if (bufferHead.getLeftLegState() == LegTweakBuffer.UNLOCKED) {
+			Vector3f leftFootDif = leftFootPosition
+				.subtract(bufferHead.getParent().getLeftFootPositionCorrected())
+				.setY(0);
+			if (leftFootDif.length() > 0.005f) {
+				float leftY = leftFootPosition.y;
+
+				Vector3f temp = bufferHead.getParent().getLeftFootPositionCorrected();
+				Vector3f velocity = bufferHead.getLeftFootVelocity();
+				// first add the difference from the last frame to this frame
+				temp = temp
+					.subtract(
+						bufferHead
+							.getParent()
+							.getLeftFootPosition()
+							.subtract(bufferHead.getLeftFootPosition())
+					);
+				leftFootPosition.y = leftY;
+				leftFootPosition.x = temp.x;
+				leftFootPosition.z = temp.z;
+
+				// if velocity and dif are pointing in the same direction,
+				// add a small amount of velocity to the dif
+				// else subtract a small amount of velocity from the dif
+				// calculate the correction weight
+				float weight = calculateCorrectionWeight(
+					leftFootPosition,
+					bufferHead.getParent().getLeftFootPositionCorrected()
+				);
+				if (velocity.x * leftFootDif.x > 0) {
+					leftFootPosition.x += velocity.x * weight;
+				} else {
+					leftFootPosition.x -= velocity.x * weight;
+				}
+				if (velocity.z * leftFootDif.z > 0) {
+					leftFootPosition.z += velocity.z * weight;
+				} else {
+					leftFootPosition.z -= velocity.z * weight;
+				}
+				// if the foot overshot the target, move it back to the target
+				if (
+					checkOverShoot(
+						this.bufferHead.getLeftFootPosition().x,
+						this.bufferHead.getParent().getLeftFootPositionCorrected().x,
+						leftFootPosition.x
+					)
+				) {
+					leftFootPosition.x = bufferHead.getLeftFootPosition().x;
+				}
+				if (
+					checkOverShoot(
+						this.bufferHead.getLeftFootPosition().z,
+						this.bufferHead.getParent().getLeftFootPositionCorrected().z,
+						leftFootPosition.z
+					)
+				) {
+					leftFootPosition.z = bufferHead.getLeftFootPosition().z;
+				}
+
+
+			}
+		}
+	}
+
+	private void correctUnlockedRightFootTracker() {
+		if (bufferHead.getRightLegState() == LegTweakBuffer.UNLOCKED) {
+			Vector3f rightFootDif = rightFootPosition
+				.subtract(bufferHead.getParent().getRightFootPositionCorrected())
+				.setY(0);
+			if (rightFootDif.length() > 0.005f) {
```

Comment:
Throughout the code I've seen this `0.005f` around, what is it? Should it be a constant instead?